test/components/collision_detection_make_activity_test.dart

```diff
@@ -0,0 +1,225 @@
+import 'dart:math';
+import 'package:flame/components.dart';
+import 'package:flame/extensions.dart';
+import 'package:flame/geometry.dart';
+import 'package:flame/src/geometry/collision_detection.dart'
+    as CollisionDetectionFunction;
+import 'package:test/test.dart';
+
+class _TestBlock extends PositionComponent with Hitbox, Collidable {
+  Set<Collidable> oncollidables = {};
+  Set<Collidable> cachecollidables = {};
+  _TestBlock() {
+    addShape(HitboxCircle());
+  }
+  @override
+  void onCollision(Set<Vector2> points, Collidable other) {
+    oncollidables.add(other);
+    super.onCollision(points, other);
+  }
+}
+
+class _TestBlockA extends _TestBlock {
+  _TestBlockA(Vector2 pos, Vector2 size) {
+    activeCollidable = false;
+    position = pos;
+    this.size = size;
+  }
+}
+
+class _TestBlockB extends _TestBlock {
+  _TestBlockB(Vector2 pos, Vector2 size) {
+    activeCollidable = false;
+    position = pos;
+    this.size = size;
+  }
+}
+
+class _TestBlockC extends _TestBlock {
+  _TestBlockC(Vector2 pos, Vector2 size) {
+    activeCollidable = true;
+    position = pos;
+    this.size = size;
+  }
+}
+
+class _TestBlockD extends _TestBlock {
+  _TestBlockD(Vector2 pos, Vector2 size) {
+    activeCollidable = true;
+    position = pos;
+    this.size = size;
+  }
+}
+
+void _testCollisionDetection(List<Collidable> collidables) {
```

Comment:
I'm not sure why this is needed? in order to test the function we should call the real function (CollisionDetectionFunction.collisionDetection) and not replicate it

Reply:
> 我不确定为什么需要这样做？为了测试该函数，我们应该调用真实函数（CollisionDetectionFunction.collisionDetection）而不是复制它

 \_Testcollisiondetection, the function is implemented from
 > https://github.com/flame-engine/flame/blob/master/lib/src/geometry/collision_ detection.dart 
 
---

\_ The testcollisiondetection function has simple and correct logic

\_Testcollisiondetection and collis ionDetectionFunction.collisionDetection

For comparison,

To keep Collis safe ionDetectionFunction.collisionDetection There will be no conflict_ Logic of different effects of test collisiondetection

Reply:
Here's (CollisionDetectionFunction.collisionDetection ）And (_ Test collisiondetection) is a different implementation

Reply:
I apologize for the code that I am confused about, I call the real function Q_Q here, the code pointed out by luanpotter comes from https://github.com/flame-engine/flame/blob/master/lib/src/geometry/collision
This is the code from flame-master that is available and does not need to be tested to ensure the correct logic

,

[CollisionDetectionFunction.collisionDetection] is the code that needs to be tested,

I use set to merge the results of the above two functions and compare the length of the set with the previous results to see if [CollisionDetectionFunction.collisionDetection] has skipped any correct collision detection